tests/test_modules.py

```diff
@@ -620,3 +620,40 @@ def __torch_function__(cls, func, types, args=(), kwargs=None):
 
         self.assertEqual(cnt.op_count, 4)
         self.assertTrue(torchdynamo.testing.same(out1, out2))
+
+    def test_torch_function_with_closure(self):
+        def run():
+
+            counter = 0
+
+            def foo(x):
+                # function call, twice to test wrapping
+                x = F.sigmoid(x)
+                x = F.sigmoid(x)
+                # method call, twice to test wrapping
+                x = x.sigmoid()
+                x = x.sigmoid()
+                return x
+
+            class TensorProxy(torch.Tensor):
+                @classmethod
+                def __torch_function__(cls, func, types, args=(), kwargs=None):
+                    nonlocal counter
+                    # for now, only support reads from closure cells
+                    # TODO(future PR): support writes as well
+                    counter + 1
```

Comment:
writing here doesn't actually crash the code, but saving the investigation to guarantee correctness of writing like this for a future PR